Snap 2014 : Column table size is wrong in UI after truncate table

snappy-sachin opened the pull request:
## Changes proposed in this pull request

- Reseting numOverflowOnDisk & numOverflowBytesOnDisk to zero in BucketRegion.updateSizeOnClearRegion method.

## Patch testing

- Tested manually

## ReleaseNotes changes

 - None

## Other PRs 

 - None

snappy-sachin committed:
SNAP-2014: Column table size is wrong in UI after truncate table.
Changes:
  - Reseting numOverflowOnDisk & numOverflowBytesOnDisk to zero in BucketRegion.updateSizeOnClearRegion method.

snappy-sachin committed:
updating comments in code.

sumwale commented on gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/BucketRegion.java:
```diff
@@ -2723,12 +2723,10 @@ else if (GemFireCacheImpl.gfxdSystem()) {
-      this.numOverflowOnDisk.set(0);
-      this.numOverflowBytesOnDisk.set(0);
-    }
+    this.numOverflowOnDisk.set(0);
```
Not sure about moving out numOverflowOnDisk too. See DiskRegion.clearStats which is called on truncate. It reduces current size (and does not zero it), so if it gets invoked after this method then it will make this negative. I had added the getDiskRegion() check for that reason but failed to notice that numOverflowBytesOnDisk is not being reduced by statsClear. Will suggest moving out just numOverflowBytesOnDisk outside of the (getDiskRegion() == null) check and keeping it for numOverflowOnDisk

sumwale replied:
Perhaps better option is to leave this code untouched, rather add call to "incNumOverflowBytesOnDisk()" in DiskRegion.statsClear with appropriate negative argument as being done for others there. This will make things uniform (like clearing this from the DiskRegion.statsClose too).

snappy-sachin replied:
As discussed, made the appropriate changes.

sumwale reviewed (commented):
see comment

snappy-sachin committed:
Merge branch 'snappy/master' into SNAP-2014

snappy-sachin committed:
SNAP-2014: Column table size is wrong in UI after truncate table
Changes:
 - Reverting code changes made previousely in commit id #e7b464842018af03f0d9c5a6f3e22a10a59af58f.
 - Resetting numOverflowBytesOnDisk in DiskRegion.statsClear() methos.

snappy-sachin commented:
As discussed, feedback comments are incorporated.
